components/tidb_query_executors/src/runner.rs

```diff
@@ -395,7 +400,7 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
         BATCH_INITIAL_SIZE
     }
 
-    pub async fn handle_request(&mut self) -> Result<SelectResponse> {
+    pub async fn handle_request(&mut self) -> Result<(SelectResponse, Option<IntervalRange>)> {
```

Comment:
Could you add some comments about the returned `IntervalRange`? For example, what does it mean, what will be the behavior if there are multiple ranges to select and whether there are constraint when multiple ranges are involved?